### ISSUE TITLE
std::bad_alloc

### ISSUE THREAD

**Issue by kristinastenlokk** [DecodeGenetics/LRcaller#1]
Hi, 

We run into erros (std::bad_alloc) while trying to run LRcaller with the following settings:

```
tools/LRcaller -nt 5 /mnt/SCRATCH/kristenl/Alignments/Barry_to_Simon.bam /mnt/SCRATCH/kristenl/SV_calling/SVIM/Barry_to_Simon/Barry_to_Simon_SVIM_q10.vcf test.vcf -v
```

Verbose:
[LRcaller_out.txt](https://github.com/DecodeGenetics/LRcaller/files/4097413/LRcaller_out.txt)

Do you have any suggestions on how to fix this?

Thanks

**Comment by bjarnivh** [comment]
The most likely explanation is that you ran out of memory when running the program.  Can you try running the program.  When running the analysis for our biorxiv manuscript we used an average 18G memory, max 41G.  You will need similar amount of memory on your machine plus the memory required for other processes running.

**Comment by kristinastenlokk** [comment]
Unfortunately, it doesn't seem to be a memory problem. We've tried running it with 2TB, without detecting any peak in memory. We've also tried to split up our vcf into smaller chuncks. Also, the alternative aligners have been tested. We get the same error: 

> terminate called after throwing an instance of 'std::bad_alloc'
>   what():  std::bad_alloc

Do you gave any other suggestions about what we can do?

**Comment by bjarnivh** [comment]
My best bet is that this has something to do with the memory allocation when there are multiple threads being used.  Can you try running using only a single thread (-nt 1) and see if you still get the error?  

I will try to debug this and see if I can reproduce the error.  If you could send me the input file the problem would be easier to debug, but I will try to recreate the error on our systems.  

Since we haven't encountered this error here, can you also tell me what OS you are running on and whether you used the precompiled binaries or compiled this yourself.

**Comment by kristinastenlokk** [comment]
I still get the same error with one thread. Our system runs on CentOS Linux 7, and I used the precompiled binaries.

I will send you a subset of our data by mail shorty.

Thanks a lot for your help.

**Comment by bjarnivh** [comment]
We identified a potential issue with the code.  Can you try downloading version 0.11 and see if the problem persists.  If this still does not solve the problem I would need to get a sample to try to debug the problem.

**Comment by bjarnivh** [comment]
The program handled hardclipped reads incorrectly.  This problem has been fixed in version 0.1.2.  We do not recommend using hardclipped reads, e.g. by enabling the -Y option in minimap2.
